crates/ruff_linter/src/rules/flake8_pyi/rules/duplicate_union_member.rs

```diff
@@ -27,63 +31,176 @@ use crate::checkers::ast::Checker;
 /// foo: str
 /// ```
 ///
+/// ## Fix safety
+/// This rule's fix is marked as safe unless the union contains comments.
+///
+/// For nested union, the fix will flatten type expressions into a single
+/// top-level union.
```

Comment:
The reasons for removing the item are different: one is because of exact duplicates, the other because `complex` implies `float` and `int`. There are cases where libs prefer being explicit and disable the second rule (stdlib random is an example). 

A dedicated rule for flatten a nested union could also work, but then should always be enabled together with these rules to clean up the fix result.